localstack/aws/protocol/serializer.py

```diff
@@ -800,6 +805,42 @@ class RestJSONResponseSerializer(BaseRestResponseSerializer, JSONResponseSeriali
     pass
 
 
+class SqsResponseSerializer(QueryResponseSerializer):
+    """
+    Unfortunately, SQS uses a rare interpretation of the XML protocol: It uses HTML entities within XML tag text nodes.
+    For example:
+    - Normal XML serializers: <Message>No need to escape quotes (like this: ") with HTML entities in XML.</Message>
+    - SQS XML serializer: <Message>No need to escape quotes (like this: &quot;) with HTML entities in XML.</Message>
+
+    None of the prominent XML frameworks for python allow HTML entity escapes when serializing XML.
+    This serializer implements the following workaround:
+    - Escape quotes and \r with their HTML entities (&quot; and &#xD;).
+    - Since & is (correctly) escaped in XML, the serialized string contains &amp;quot; and &amp;#xD;
+    - These double-escapes are corrected by replacing such strings with their original.
+    """
+
+    def _default_serialize(self, xmlnode: ETree.Element, params: str, _, name: str) -> None:
+        """Ensures that XML text nodes use HTML entities instead of " or \r"""
+        node = ETree.SubElement(xmlnode, name)
+        node.text = six.text_type(params).replace('"', "&quot;").replace("\r", "&#xD;")
```

Comment:
For now I just handled the double quotes and the carriage return, since these are the ones handled in our previous SQS provider implementation and moto.
What's your opinion, should we directly use the html escaping logic to handle all HTML-forbidden characters? The downside would be that we would have to handle them individually when reverting the double-escaping (in the function below).

Reply:
no i think this looks good to me for now!